code/datums/components/chasm.dm

```diff
@@ -86,58 +88,147 @@
+	dropped_thing.visible_message(span_boldwarning("[dropped_thing] falls into [parent]!"), span_userdanger("[oblivion_message]"))
+	if (isliving(dropped_thing))
+		var/mob/living/falling_mob = dropped_thing
+		falling_mob.notransform = TRUE
+		falling_mob.Paralyze(20 SECONDS)
 
+	var/oldtransform = dropped_thing.transform
+	var/oldcolor = dropped_thing.color
+	var/oldalpha = dropped_thing.alpha
+
+	animate(dropped_thing, transform = matrix() - matrix(), alpha = 0, color = rgb(0, 0, 0), time = 10)
+	for(var/i in 1 to 5)
 		//Make sure the item is still there after our sleep
-		if(!AM || QDELETED(AM))
+		if(!dropped_thing || QDELETED(dropped_thing))
 			return
+		dropped_thing.pixel_y--
+		sleep(0.2 SECONDS)
 
-		if(iscyborg(AM))
-			var/mob/living/silicon/robot/S = AM
-			qdel(S.mmi)
-
-		falling_atoms -= AM
-		qdel(AM)
-		if(AM && !QDELETED(AM))	//It's indestructible
-			var/atom/parent = src.parent
-			parent.visible_message(span_boldwarning("[parent] spits out [AM]!"))
-			AM.alpha = oldalpha
-			AM.color = oldcolor
-			AM.transform = oldtransform
-			AM.throw_at(get_edge_target_turf(parent,pick(GLOB.alldirs)),rand(1, 10),rand(1, 10))
+	//Make sure the item is still there after our sleep
+	if(!dropped_thing || QDELETED(dropped_thing))
+		return
+
+	if (!storage)
+		storage = new(get_turf(parent))
+		RegisterSignal(storage, COMSIG_ATOM_EXITED, .proc/left_chasm)
+
+	if (storage.contains(dropped_thing))
+		return
+
+	dropped_thing.alpha = oldalpha
+	dropped_thing.color = oldcolor
+	dropped_thing.transform = oldtransform
+
+	if (dropped_thing.forceMove(storage))
+		if (isliving(dropped_thing))
+			RegisterSignal(dropped_thing, COMSIG_LIVING_REVIVE, .proc/on_revive)
+	else
+		parent.visible_message(span_boldwarning("[parent] spits out [dropped_thing]!"))
+		dropped_thing.throw_at(get_edge_target_turf(parent, pick(GLOB.alldirs)), rand(1, 10), rand(1, 10))
+
+	if (isliving(dropped_thing))
+		var/mob/living/fallen_mob = dropped_thing
+		if(fallen_mob.stat != DEAD)
+			fallen_mob.death(TRUE)
+			fallen_mob.notransform = FALSE
+			fallen_mob.apply_damage(300)
+			var/obj/item/bodypart/l_leg = fallen_mob.get_bodypart(BODY_ZONE_L_LEG)
+			var/datum/wound/blunt/critical/l_fracture = new
+			l_fracture.apply_wound(l_leg)
+			var/obj/item/bodypart/r_leg = fallen_mob.get_bodypart(BODY_ZONE_R_LEG)
+			var/datum/wound/blunt/critical/r_fracture = new
+			r_fracture.apply_wound(r_leg)
+
+	falling_atoms -= falling_ref
+
+/**
+ * Called when something has left the chasm depths storage.
+ * Arguments
+ *
+ * * source - Chasm object holder.
+ * * gone - Item which has just left the chasm contents.
+ */
+/datum/component/chasm/proc/left_chasm(atom/source, atom/movable/gone)
+	SIGNAL_HANDLER
+	UnregisterSignal(gone, COMSIG_LIVING_REVIVE)
+
+#define CHASM_TRAIT "chasm trait"
+
+/**
+ * Called if something comes back to life inside the pit. Expected sources are badmins and changelings.
+ * Ethereals should take enough damage to be smashed and not revive.
```

Comment:
this comment should probably be removed, our ethereals don't do this

Reply:
Ethereals don't shatter?

Reply:
This is referring to them self reviving using their crystal core. Ethereals on are server are closer slimes than crystals